### PR TITLE
Added offset and sorting and nextPageURL

### PR DESCRIPTION
**Pull request by mutantkeyboard**
In this commit we added:
- Offset
- Sorting (by various conditions)
- nextPageURL
- More tests

TODO: 
- Add more tests for edge cases
- Add caching

### CODE REVIEW

File: util.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package fiberpaginate
+
+import "strings"
+
+/* This file contains all the utility and helper functions that can be reused in paginate */
+
+// isAllowedSort returns true if the given sort string is allowed in the given list of allowedSorts,
+// false otherwise.
+func isAllowedSort(sort string, allowedSorts []string) bool {
```
Comment (garrettladley) [inline util.go]: is this redundant if the stdlib has [`slices.Contains`](https://pkg.go.dev/slices#Contains)

File: util.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+// If a field is not allowed, it is skipped.
+// The order of each field is determined by its prefix, with "-" indicating DESC and no prefix indicating ASC.
+// If no allowed fields are found, the same single-element slice is returned with the defaultSort.
+func parseSortQuery(query string, allowedSorts []string, defaultSort string) []SortField {
+	if query == "" {
+		return []SortField{{Field: defaultSort, Order: ASC}}
+	}
+
+	fields := strings.Split(query, ",")
+	var sortFields []SortField
```
Comment (garrettladley) [inline util.go]: since the len of `sortFields` is directly proportional to `len(fields)`, can we allocate a slice of `len(fields)` here?

File: util.go
Comment (garrettladley) [inline util.go:1]: nit: i don't _love_ `util.go` as a file name, `strings.go` seems more fitting based on the contents

File: util.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	return false
+}
+
+// parseSortQuery takes a query string and a list of allowed sorts, and returns a slice of SortFields.
+// If the query string is empty, it returns a slice with a single SortField with the given defaultSort.
+// The query string is split on commas, and each field is checked against the allowedSorts.
+// If a field is not allowed, it is skipped.
+// The order of each field is determined by its prefix, with "-" indicating DESC and no prefix indicating ASC.
+// If no allowed fields are found, the same single-element slice is returned with the defaultSort.
+func parseSortQuery(query string, allowedSorts []string, defaultSort string) []SortField {
```
Comment (garrettladley) [inline util.go]: could we add some tests for this?

**Review by garrettladley — CHANGES_REQUESTED** [review]

**Comment by mutantkeyboard** [comment]
@garrettladley - do I close a PR or just leave it open?
